BlueDBOnDisk/src/main/java/io/bluedb/disk/segment/SegmentManager.java

```diff
@@ -0,0 +1,164 @@
+package io.bluedb.disk.segment;
+
+import java.io.File;
+import java.io.Serializable;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import io.bluedb.api.keys.BlueKey;
+import io.bluedb.api.keys.TimeFrameKey;
+import io.bluedb.disk.collection.BlueCollectionImpl;
+
+public class SegmentManager<T extends Serializable> {
+
+	protected static final String SEGMENT_SUFFIX = "_seg";
+	// TODO better split up
+	protected static final long LEVEL_3 = TimeUnit.HOURS.toMillis(1);
+	protected static final long LEVEL_2 = TimeUnit.DAYS.toMillis(1);
+	protected static final long LEVEL_1 = LEVEL_2 * 30;
+	protected static final long LEVEL_0 = LEVEL_1 * 12;
+
+	private final BlueCollectionImpl<T> collection;
+	
+	public SegmentManager(BlueCollectionImpl<T> collection) {
+		this.collection = collection;
+	}
+
+	public Segment<T> getFirstSegment(BlueKey key) {
+		long groupingNumber = key.getGroupingNumber();
+		Path segmentPath = getPath(groupingNumber);
+		return toSegment(segmentPath);
+	}
+
+	public List<Segment<T>> getAllSegments(BlueKey key) {
+		return getAllPossibleSegmentPaths(key).stream()
+				.map((p) -> (toSegment(p)))
+				.collect(Collectors.toList());
+	}
+
+	public List<Segment<T>> getExistingSegments(long minValue, long maxValue) {
+		return getExistingSegmentFiles(minValue, maxValue).stream()
+				.map((f) -> (toSegment(f.toPath())))
+				.collect(Collectors.toList());
+	}
+
+	protected Path getPath(BlueKey key) {
+		long groupingNumber = key.getGroupingNumber();
+		return getPath(groupingNumber);
+	}
+
+	protected Path getPath(long groupingNumber) {
+		String level0 = getRangeFileName(groupingNumber, LEVEL_0);
+		String level1 = getRangeFileName(groupingNumber, LEVEL_1);
+		String level2 = getRangeFileName(groupingNumber, LEVEL_2);
+		String level3 = getRangeFileName(groupingNumber, LEVEL_3);
+		return Paths.get(collection.getPath().toString(), level0, level1, level2, level3 + SEGMENT_SUFFIX);

```

Comment:
I'm still a little confused how this works. Maybe we could white board it or something tomorrow. I think looking at it in eclipse will be a bit easier.